src/features/auth/ui/PetInformationForm.tsx

```diff
@@ -231,36 +267,46 @@ export const CharacterInput = () => {
             key={idx}
             label={personality}
             onClick={() => setPersonalities(personality)}
+            uncontrolledInitialIsSelected={initialPersonalities.includes(
+              personality,
+            )}
           />
         ))}
       </div>
     </div>
   );
 };
 
-export const IntroduceTextArea = () => {
-  const setDescription = usePetInfoStore((state) => state.setDescription);
+const PetDescriptionTextArea = () => {
+  const { description } = usePetInformationFormContext().getState();
+  const setDescription = usePetInformationFormStore(
+    (state) => state.setDescription,
+  );
   return (
     <TextArea
       id="introduce"
       label="간단히 소개해 주세요"
       placeholder="우리 댕댕이를 간단히 소개해주세요"
       statusText=""
       onChange={(e) => setDescription(e.target.value)}
+      defaultValue={description}
+      initialCurrentLength={description.length}
     />
   );
 };
 
-export const SubmitButton = () => {
-  const { mutate: postPetInfo } = usePostPetInfo();
-
+const SubmitButton = ({
+  onSubmit,
+  disabled,
+}: Omit<PetInformationFormProps, "initialState">) => {
+  const store = usePetInformationFormContext();
   // 필수 항목을 모두 입력하지 않은 경우 나타 날 스낵바
   const { handleOpen: openInfoSnackBar, onClose } = useSnackBar(() => (
     <Snackbar onClose={onClose}>필수 항목을 모두 입력해 주세요</Snackbar>
   ));
 
   const handleClick = () => {
-    const petInfoForm = usePetInfoStore.getState();
+    const petInfoForm = store.getState();
```

Comment:
사소한거지만 아래처럼 가져와도 될 것 같아요!
다른 곳도 다 아래 방법으로 가져오고 있어서요!

```suggestion
const petInfoForm = usePetInformationFormContext().getState();
```

Reply:
훅은 핸들러 내부에서 호출 불가해 `store` 를 컴포넌트에서 불러온 후 가져왔습니다

Reply:
제가 잘못봤네요! 확인했습니다